spec/models/service/retirement_management_spec.rb

```diff
@@ -7,10 +7,10 @@
 
   # shouldn't be running make_retire_request because it's the bimodal not from ui part
   it "#retirement_check" do
+    User.current_user = user
```

Comment:
@d-m-u 
Can we use User.with_user with a do block similar to this
https://github.com/ManageIQ/manageiq/blob/a9ea2507e0ff7b2939441f0a3be7bfbf4996f362/spec/lib/rbac/filterer_spec.rb#L263